apple/Tests/TestCases/Profile/AppPreferencesTests.swift

```diff
@@ -17,8 +17,26 @@ final class AppPreferencesTests: Test<AppPreferences> {
 	}
 	
 	func test_has_gateway() {
-		XCTAssertTrue(SUT.default.hasGateway(with: .init(string: "https://mainnet.radixdlt.com/")!))
-		XCTAssertTrue(SUT.default.hasGateway(with: .init(string: "https://mainnet.radixdlt.com")!))
-		XCTAssertFalse(SUT.default.hasGateway(with: .init(string: "https://radixdlt.com")!))
+		// Valid URLs
+		do {
+			var result = try SUT.default.hasGateway(with: .init(string: "https://mainnet.radixdlt.com/")!)
+			XCTAssertTrue(result)
+			result = try SUT.default.hasGateway(with: .init(string: "https://mainnet.radixdlt.com")!)
+			XCTAssertTrue(result)
+			result = try SUT.default.hasGateway(with: .init(string: "https://radixdlt.com")!)
+			XCTAssertFalse(result)
+		} catch {
+			XCTFail("Unexpected failure")
+		}
+		
+		// Invalid URL
+		do {
```

Comment:
and I would prefer if this expected throws to be its own test, and use `XCTAssertThrows`